modem: ubx: fix incoming byte processing

joerchan opened the pull request:
`ret` holds the amount of bytes received from the modem. However during processing of the bytes its value is overwritten by the return value of `modem_ubx_process_received_byte`, in practice discarding all but the first byte read.

To prevent this, store the length in a separate variable.

Handle malloc returning NULL pointer, set err and return from function.

andak committed:
modem: ubx: fix incoming byte processing

`ret` holds the amount of bytes received from the modem. However during
processing of the bytes its value is overwritten by the return value of
`modem_ubx_process_received_byte`, in practice discarding all but the
first byte read.

To prevent this, store the length in a separate variable.

Signed-off-by: Anders T. Akre <anders@akre.io>

tomi-font commented on subsys/modem/modem_ubx.c:
```diff
@@ -269,7 +269,8 @@ static void modem_ubx_process_handler(struct k_work *item)
 	}
 
-	for (int i = 0; i < ret; i++) {
+	const size_t length = ret;
```
Fix for compliance:
```suggestion
	const size_t length = ret;

```

joerchan replied:
That was quick :) Thanks.

joerchan committed:
drivers: gnss: Do not re-assign err from helper function

Do not re-assign err from helper function, this just causes loss of
information from the helper function.
Only possible err returned is -EINVAL, so not a functional change.

Signed-off-by: Joakim Andersson <joerchan@gmail.com>

joerchan committed:
drivers: gnss: Handle malloc returning NULL pointer

Handle malloc returning NULL pointer, set err and return from function.

Signed-off-by: Joakim Andersson <joerchan@gmail.com>

joerchan commented:
Found an unhandled possible malloc null-pointer ref.

tomi-font commented on drivers/gnss/gnss_u_blox_m10.c:
Nit in the commit message: `this just reduces causes loss of information`

joerchan replied:
Fixed, thanks.
